packages/iris-grid/src/IrisGridProxyModel.ts

```diff
@@ -0,0 +1,610 @@
+            detail: err,
+          })
+        );
+      });
+  }
+
+  startListening(): void {
+    super.startListening();
+
+    this.addListeners(this.model);
+  }
+
+  stopListening(): void {
+    super.stopListening();
+
+    this.removeListeners(this.model);
+  }
+
+  addListeners(model: IrisGridTreeTableModel | IrisGridTableModel): void {
+    const events = Object.keys(IrisGridModel.EVENT);
+    for (let i = 0; i < events.length; i += 1) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      model.addEventListener(events[i], this.handleModelEvent);
+    }
+  }
+
+  removeListeners(model: IrisGridTreeTableModel | IrisGridTableModel): void {
+    const events = Object.keys(IrisGridModel.EVENT);
+    for (let i = 0; i < events.length; i += 1) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      model.removeEventListener(events[i], this.handleModelEvent);
+    }
+  }
+
+  get rowCount(): number {
+    return this.model.rowCount;
+  }
+
+  get columnCount(): number {
+    return this.model.columnCount;
+  }
+
+  get floatingTopRowCount(): number {
+    return this.model.floatingTopRowCount;
+  }
+
+  get floatingBottomRowCount(): number {
+    return this.model.floatingBottomRowCount;
+  }
+
+  get floatingLeftColumnCount(): number {
+    return this.model.floatingLeftColumnCount;
+  }
+
+  get floatingRightColumnCount(): number {
+    return this.model.floatingRightColumnCount;
+  }
+
+  textForCell: IrisGridTableModel['textForCell'] = (...args) =>
+    this.model.textForCell(...args);
+
+  truncationCharForCell: IrisGridTableModel['truncationCharForCell'] = (
+    ...args
+  ) => this.model.truncationCharForCell(...args);
+
+  textAlignForCell: IrisGridTableModel['textAlignForCell'] = (...args) =>
+    this.model.textAlignForCell(...args);
+
+  colorForCell: IrisGridTableModel['colorForCell'] = (...args) =>
+    this.model.colorForCell(...args);
+
+  backgroundColorForCell: IrisGridTableModel['backgroundColorForCell'] = (
+    ...args
+  ) => this.model.backgroundColorForCell(...args);
+
+  textForColumnHeader: IrisGridTableModel['textForColumnHeader'] = (...args) =>
+    this.model.textForColumnHeader(...args);
+
+  textForRowHeader: IrisGridTableModel['textForRowHeader'] = (...args) =>
+    this.model.textForRowHeader(...args);
+
+  textForRowFooter: IrisGridTableModel['textForRowFooter'] = (...args) =>
+    this.model.textForRowFooter(...args);
+
+  isRowMovable: IrisGridTableModel['isRowMovable'] = (...args) =>
+    this.model.isRowMovable(...args);
+
+  isColumnMovable: IrisGridTableModel['isColumnMovable'] = (...args) =>
+    this.model.isColumnMovable(...args);
+
+  isColumnFrozen(x: number): boolean {
+    if (TableUtils.isTreeTable(this.table)) {
+      throw new Error('TreeTable cannot be frozen');
+    }
+    return (this.model as IrisGridTableModel).isColumnFrozen(x);
```

Comment:
`isColumnFrozen` should be defined on `IrisGridModel`, and then `IrisGridTreeTable` model can just return `false`.